src/GovUk.Education.ExploreEducationStatistics.Content.Model/Repository/ReleaseDataFileRepository.cs

```diff
@@ -58,19 +58,25 @@ public ReleaseDataFileRepository(ContentDbContext contentDbContext)
                     CreatedById = createdById,
                     RootPath = releaseVersionId,
                     SubjectId = subjectId,
+                    DataSetFileId = type != Data
+                        ? null
+                        : replacingDataFile?.DataSetFileId ?? Guid.NewGuid(),
```

Comment:
I think if I made this change, it would throw this exception every time you imported a new data set.

If you're importing a new data set, then `type == Data && replacingDataFile == null`, so it would be null and the new guid / version 0 get set.